core/engine/src/builtins/error/mod.rs

```diff
@@ -130,20 +130,13 @@ pub enum ErrorKind {
 /// [`JsNativeErrorKind`][crate::error::JsNativeErrorKind].
 ///
 /// [spec]: https://tc39.es/ecma262/#sec-error-objects
-#[derive(Debug, Clone, Trace, Finalize, JsData)]
+#[derive(Debug, Clone, PartialEq, Eq, Trace, Finalize, JsData)]
 pub struct Error {
     pub(crate) tag: ErrorKind,
-    #[unsafe_ignore_trace]
-    pub(crate) position: Option<ShadowEntry>,
-}
 
-impl Eq for Error {}
-impl PartialEq for Error {
     // The position of where the Error was created does not affect equality check.
-    #[inline]
-    fn eq(&self, other: &Self) -> bool {
-        self.tag == other.tag
-    }
+    #[unsafe_ignore_trace]
+    pub(crate) position: IgnoreEq<Option<ShadowEntry>>,
```

Comment:
`IgnoreEq` looks practical, but it is only used once, so I don't think it's better than just implementing `PartialEq` manually. This also avoids having a type that technically implements `Eq` in an "incorrect" way.

Reply:
Oh, didn't consider that you could implement an empty hash. Yeah, that solves all my issues with this